twitter-core/src/main/java/com/twitter/sdk/android/core/identity/TwitterAuthClient.java

```diff
@@ -109,13 +117,27 @@ public void cancelAuthorize() {
     }
 
     private boolean authorizeUsingSSO(Activity activity, CallbackWrapper callbackWrapper) {
-        if (SSOAuthHandler.isAvailable(activity)) {
-            Twitter.getLogger().d(TwitterCore.TAG, "Using SSO");
-            return authState.beginAuthorize(activity,
-                    new SSOAuthHandler(authConfig, callbackWrapper, authConfig.getRequestCode()));
-        } else {
-            return false;
-        }
+        Twitter.getLogger().d(TwitterCore.TAG, "Using SSO");
+        Callback<TwitterSession> ssoCallbackWrapper = new Callback<TwitterSession>() {
+            @Override
+            public void success(Result<TwitterSession> result) {
+                callbackWrapper.success(result);
+            }
+
+            @Override
+            public void failure(TwitterException exception) {
+                if (exception instanceof TwitterSSOAuthException) {
+                    authState.endAuthorize();
+                    if (!authorizeUsingOAuth(activity, callbackWrapper)) {
+                        callbackWrapper.failure(new TwitterAuthException("Authorize failed."));
+                    }
+                    return;
+                }
+                callbackWrapper.failure(exception);
+            }
+        };
+        return authState.beginAuthorize(activity,
+                new SSOAuthHandler(authConfig, ssoCallbackWrapper, authConfig.getRequestCode()));
```

Comment:
Nit: why need to intercept this & return new TwitterAuthException() again?

Reply:
The `TwitterSSOAuthException` is for checking the null token / null token secret issue. 
When encountering this kind of issue, we go to OAuth authorization again via web view which I use `ssoCallbackWrapper` for doing this. If there's any failure inside the OAuth authorization, we still need to return `TwitterAuthException`.